modules/virt-creating-a-service-backed-by-virtual-machines.adoc

```diff
@@ -0,0 +1,142 @@
+
+To create a service backed by a virtual machine, you start the `VirtualMachineInstance`,
+if it is not already running. Once the instance is running, you connect to the instance, and
+create a service object to expose the instance.
+
+The following service types are supported:
+
+* `ClusterIP` - Connect using service IP and service port inside the cluster network.
+* `NodePort` - Connect using node IP and node port outside the cluster network.
+* `LoadBalancer` - Connect using public IP and port.
+
+This procedure presents an example of how to create and expose a service object of type `ClusterIP`
+as a virtual machine-backed service.
+
+[NOTE]
+====
+Labels on a `VirtualMachineInstance` are passed through to the pod. The labels on
+the `VirtualMachineInstance` must match the labels on the Service object.
+====
+
+.Procedure
+
+. Edit the `VirtualMachineInstance` YAML configuration file to include the
+label `special: key:`.
++
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachineInstance
+metadata:
+  name: vmi-ephemeral
+  labels:
+    special: key <1>
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+    resources:
+      requests:
+        memory: 64M
+  volumes:
+  - name: containerdisk
+    containerDisk:
+      image: kubevirt/cirros-registry-disk-demo:latest
+----
+<1> The `special: key:` label
++
+
+. Edit the YAML to create and expose the Service object.
+
+.. Add `type: ClusterIP` for a
+ClusterIP service. Add `targetPort: 22`, exposing the service on SSH port `22`.
++
+To create and expose other types of services, such as `NodePort` and `LoadBalancer`, replace
+`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate,
+in the Service object definition.
++
+.. Choose a descriptive `name` for the service you are creating and exposing.
+Unlike `type`, which must contain specific values, you can choose any descriptive name
+for the service. In this example, note that the `name` is `vmiservice`, but for a
+node port service, you might choose the name `nodeport`. For a load balancer service,
+you might choose the name `lbsvc`.
++
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmiservice
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 22 <2>
+  selector:
+    special: key
+  type: ClusterIP <1>
+----
+<1> Defines Service `type: ClusterIP`
+<2> Exposes SSH `targetPort` 22
++
+
+. Create the `ClusterIP` service using `kubectl` with the service `name` you used when defining
+the Service object. In this example, the `<service_name>` is `vmiservice`,
+as you defined it in the Service object YAML configuration file.
++
+
+[source,terminal]
+----
+$ kubectl create -f <service_name>.yaml
+----
++
+
+. Query the Service object to verify it is available and is configured with type `ClusterIP`.
++
+
+[source, terminal]
+----
+$ kubectl get service
```

Comment:
The service is namespace scoped. Not sure we want to use a 'particular' namespace throughout the example to make that relationship explicit.

Just a thought.

Reply:
+1

Reply:
Good thought. @maiqueb and @rhrazdil - Can you give an example of the YAML with namespace defined? 
Do we need to specify it elsewhere?

I can add a NOTE to remind the user that the service must be defined to the namespace (if that's what you mean by namespace scoped).

Reply:
I have added the following:

Because the service is scoped by the namespace to which it is defined, ensure 
that you reference the correct namespace in all YAML definitions.

I am tagging @phoracek for verification and also sent him email to ensure I have interpreted this feedback correctly.

Reply:
Reworded the namespace statement above and added a new question to PR. Resolving this as it is obsolete.